image_transport/include/image_transport/image_transport.hpp

```diff
@@ -109,9 +107,13 @@ class ImageTransport
   using ImageConstPtr = sensor_msgs::msg::Image::ConstSharedPtr;
   using CameraInfoConstPtr = sensor_msgs::msg::CameraInfo::ConstSharedPtr;
 
+  [[deprecated("Use ImageTransport(image_transport::RequiredInterfaces node_interfaces) instead")]]
   IMAGE_TRANSPORT_PUBLIC
   explicit ImageTransport(rclcpp::Node::SharedPtr node);
```

Comment:
Is this deprecation really needed / desired? I would assume most users are only interested in equipping their (lifecycle) nodes with the respective publishers / subscribers for e.g. compressed image transport. Consequently, I don't see a benefit of forcing people to know more about the internals of the package. 
Due to the implicit conversion behavior of `rclcpp::NodeInterfaces` the new constructor taking the `image_transport::RequiredInterfaces` already covers node instances (also for lifecycle nodes). So the only thing missing for preserving the current interface are constructors for raw and shared_ptrs, which can be quickly realized using templates and simply need to delegate to the "main" constructor.